core/src/test/scala/unit/kafka/server/ProduceRequestTest.scala

```diff
@@ -88,6 +88,19 @@ class ProduceRequestTest extends BaseRequestTest {
     assertEquals(-1, partitionResponse.logAppendTime)
   }
 
+  @Test
+  def shouldBeTransactionalWhenRecordsAreTransactional(): Unit = {
+    val (partition, leader) = createTopicAndFindPartitionWithLeader("topic")
+    val memoryRecords = MemoryRecords.withTransactionalRecords(0, CompressionType.NONE, 1L, 1.toShort, 1,
+      new SimpleRecord(System.currentTimeMillis(), "key".getBytes, "value".getBytes))
+
+    val topicPartition = new TopicPartition("topic", partition)
+    val partitionRecords = Map(topicPartition -> memoryRecords)
+
+    val request = new ProduceRequest.Builder(RecordBatch.CURRENT_MAGIC_VALUE, -1, 3000, partitionRecords.asJava).build()
+    assertTrue(request.isTransactional)
```

Comment:
Shouldn't this test be in the Java unit test that you introduced? This class is for tests that actually send a request. Or did you mean to send a request?